DOCS-3930 Moving Around the Synthetic Testing Tunnel (#15316)
DataDog/documentation: content/en/continuous_testing/_index.md

---
title: Continuous Testing
kind: documentation

> description: With test automation, use fast, codeless, and reliable testing in your CI pipelines to ship features with confidence.

Reviewer: ```suggestion
description: With test automation, use fast, codeless, and reliable testing in your CI/CD pipelines to ship features with confidence.
```

disable_sidebar: true
further_reading:
- link: 'https://learn.datadoghq.com/courses/synthetic-tests-ci-cd-pipeline'
  tag: 'Learning Center'

>   text: 'Synthetic Tests in a CI/CD Pipeline'

Reviewer: ```suggestion
  text: 'Continuous Testing in a CI/CD Pipeline'
```

- link: '/continuous_testing/troubleshooting/'
  tag: 'Documentation'

>   text: 'Synthetics and CI/CD Troubleshooting'

Reviewer: ```suggestion
  text: 'Continuous Testing and CI/CD Troubleshooting'
```

---

> With **end-to-end testing in production and CI environments**, your development teams can proactively ensure that defective code does not impact your users' experience. 

Reviewer: ```suggestion
With **integration and end-to-end testing in production and CI environments**, your development teams can proactively ensure that defective code does not impact your users' experience. 
```

You can run Synthetic tests on a CI pipeline in the [Datadog application][1], with the [API][2], or with [Terraform][2].

## Run tests with your integration and deployment processes

> Leverage your Synthetic tests as [canary deployments][3] or run them directly in your [CI pipelines][3].

Reviewer: ```suggestion
Leverage your Synthetic tests as [integration and end-to-end tests][3] for your staging, pre-prod or canary deployments, or run them directly in your [CI pipelines][3].
```

{{< img src="synthetics/ci.png" alt="CI tests" style="width:100%;">}}

## Examine test results in your CI environment

> Explore browser tests running in a CI pipeline and troubleshoot failing test results by looking at executed jobs in the [CI Results Explorer][4].

Reviewer: ```suggestion
Explore integration and end-to-end tests running in a CI/CD pipeline and troubleshoot failing test results by looking at executed jobs in the [CI Results Explorer][4].
```

> {{< img src="synthetics/ci_results_explorer/ci_results_explorer_1.png" alt="CI Results Explorer" style="width:100%;">}}

Reviewer: Could we update this image with this: https://a.cl.ly/Kou1lnBQ

Thanks to a broad support of diverse protocols, frameworks, and APIs—including gRPC and WebSockets—you can test across every level of your application stack and earlier.

## Improve test reliability 

Instead of having to implement test code, you can build software using [Synthetic Monitoring's resilient, scalable, and codeless tests][3]. Gain confidence in your test results by minimizing false positives through self-healing browser tests and automatic test retries. 

To ensure your users have the best experience, automate cross-browser testing.

## Increase efficiency through seamless integrations

Fast-track your application development by testing and troubleshooting in one platform. Use integrations with CI providers such as GitHub, GitLab, Jenkins, CircleCI, and Azure DevOps, and collaboration tools such as Slack and Jira to merge workflows and prevent context-switching. 

You can use the [Datadog Terraform provider][4] to control test creation and state management. Leverage your Synthetic tests as [integration and end-to-end tests][5] for your staging, pre-prod, and canary deployments, or run them directly in your [CI pipelines][5].

## Accelerate troubleshooting

> Performing tests in a unified monitoring platform allows you to quickly find the root cause of failed test runs and reduce MTTR. You can obtain the full context for troubleshooting—without switching between tools— through correlated metrics, traces, and logs surfaced by the Datadog APM integration. 

Author: ```suggestion
Performing tests in a unified monitoring platform allows you to quickly find the root cause of failed test runs and reduce MTTR. You can obtain the full context for troubleshooting—without switching between tools— through correlated metrics, traces, and logs surfaced by the Datadog [APM integration][8]. 
```

Enable developers to "shift left" and implement testing as part of the CI/CD process, which improves efficiency, release velocity, and business agility. Explore browser tests running in a CI pipeline and troubleshoot failing test results by looking at executed jobs in the [Explorer][6].

{{< img src="continuous_testing/explorer_ci_batches.jpg" alt="CI batches in the Continuous Testing Explorer" style="width:100%;">}}

## Ready to start?

Once you have configured some [Synthetic tests][3], see the documentation for your preferred [CI/CD provider][1] or use the [datadog-ci NPM package][7] in your CI/CD pipelines. Then, start exploring details about your batch runs in the [Explorer][5].

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /continuous_testing/cicd_integrations/
[2]: /synthetics/browser_tests
[3]: /synthetics/
[4]: https://registry.terraform.io/providers/DataDog/datadog/latest/
[5]: /continuous_testing/explorer
[6]: https://app.datadoghq.com/synthetics/create#

> [7]: /continuous_testing/cicd_integrations/configuration

Author: ```suggestion
[7]: /continuous_testing/cicd_integrations/configuration
[8]: /synthetics/apm/
```

---
title: Continuous Testing
kind: documentation

> description: With Continuous Testing, use fast, codeless, and reliable testing in your CI/CD pipelines to ship features with confidence.

Author: ```suggestion
description: Set parallelization settings for Synthetic tests running in CI/CD pipelines.
```
